internal/pkg/agent/cmd/otel.go

```diff
@@ -81,6 +84,16 @@ func RunCollector(cmdCtx context.Context, configFiles []string, supervised bool,
 	if err != nil {
 		return fmt.Errorf("failed to prepare collector settings: %w", err)
 	}
+
+	// create and start a new server that will listen for the diagnostics requests.
+	server := otel.NewServer()
```

Comment:
Should this really be started here or should it be encapsulated into an extension that is started by the collector runtime?

Reply:
+1, similar comment thread in https://github.com/elastic/elastic-agent/pull/10052#discussion_r2364425869

Start simpler and we can add on a control socket style implementation if there's a need for it later.